src/decorators/skinnable.js

```diff
@@ -31,8 +33,21 @@ export default function skinnable(template?: Function): Function {
       Component.prototype.getLocals = defaultGetLocals;
     }
 
+    const originalCWU = Component.prototype.componentWillUpdate;
+
+    Component.prototype.componentWillUpdate = function(nextProps, nextState) {
+      this._refreshLocals = (
+        !shallowEqual(omit(nextProps, 'children'), omit(this.props, 'children')) ||
+        !shallowEqual(nextState, this.state)
+      );
+      originalCWU && originalCWU.call(this, nextProps, nextState);
+    }
+
     Component.prototype.render = function () {
-      return this.template(this.getLocals(this.props));
+      if (this._refreshLocals || !this._locals) {
```

Comment:
ok no nevermind, it is alway set to the correct value at every `componentWillUpdate`